integration/nwo/fabric/network/checks.go

```diff
@@ -0,0 +1,218 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package network
+
+import (
+	"github.com/hyperledger-labs/fabric-smart-client/integration/nwo/api"
+	"github.com/hyperledger-labs/fabric-smart-client/integration/nwo/fabric/opts"
+	"github.com/hyperledger-labs/fabric-smart-client/integration/nwo/fabric/topology"
+	"github.com/hyperledger-labs/fabric-smart-client/integration/nwo/fsc"
+	. "github.com/onsi/gomega"
+)
+
+// CheckTopology checks the topology of the network
+func (n *Network) CheckTopology() {
+	if n.Templates == nil {
+		n.Templates = &topology.Templates{}
+	}
+	if n.Logging == nil {
+		n.Logging = &topology.Logging{
+			Spec:   "debug",
+			Format: "'%{color}%{time:2006-01-02 15:04:05.000 MST} [%{module}] %{shortfunc} -> %{level:.4s} %{id:03x}%{color:reset} %{message}'",
+		}
+	}
+
+	n.CheckTopologyOrderers()
+	n.CheckTopologyOrgs(n.CheckTopologyFSCNodes())
+	n.CheckTopologyFabricPeers()
+	n.CheckTopologyExtensions()
+}
+
+// CheckTopologyOrderers checks that the orderers' ports are allocated
+func (n *Network) CheckTopologyOrderers() {
```

Comment:
If I understand it correctly, this method doesn't check, but rather reserves ports.

Reply:
yes.